src/main/kotlin/org/elm/lang/core/psi/ElmPsiFactory.kt

```diff
@@ -105,22 +105,23 @@ class ElmPsiFactory(private val project: Project) {
                     ?: error("Invalid upper-case identifier: `$text`")
 
     fun createUpperCaseQID(text: String): ElmUpperCaseQID =
-            createFromText<ElmModuleDeclaration>("module $text exposing (..)")
-                    ?.upperCaseQID
+            createFromText("module $text exposing (..)")
                     ?: error("Invalid upper-case QID: `$text`")
 
     fun createValueQID(text: String): ElmValueQID =
-            createFromText<ElmValueDeclaration>("f = $text")
-                    ?.expression
-                    ?.childOfType()
+            createFromText("f = $text")
```

Comment:
There are some unrelated cleanups in this file. I can revert them if you'd like.

Reply:
sorry, I touched this in my PR. Can you rebase?

Reply:
done